tests/oper.rs

```diff
@@ -271,6 +271,45 @@ fn fold_and_sum() {
     }
 }
 
+#[test]
+fn fold_and_prod() {
+    let a = Array::linspace(0., 127., 128).into_shape((8, 16)).unwrap();
```

Comment:
The array of values shouldn't include zero. (The problem is that a product involving zero is always zero; it isn't a great test for arbitrary values. You could have a separate test involving zero, though.) After fixing this, you might also have to make the numbers smaller to keep the product from becoming too large.

Reply:
Good catch. After changing this I ran into another issue: multiplying 128 not-close-to-1 values results in a huge number that suffers from severe precision loss, so the order of multiplication becomes relevant, and the `fold` no longer yields something similar to `unrolled_prod`. I changed the testcase to `Array::linspace(0.5,2.0,128)` which seems a more appropriate use for `scalar_prod` anyway.